src/main/java/com/autotune/database/dao/ExperimentDAOImpl.java

```diff
@@ -75,6 +78,43 @@ public ValidationOutputData addExperimentToDB(KruizeExperimentEntry kruizeExperi
         return validationOutputData;
     }
 
+    @Override
+    public void deletePartitions(int daysCount) {
+        LOGGER.info("Threshold is set to {}", daysCount);
+        // Calculate the date 'daysCount' days ago
+        Calendar calendar = Calendar.getInstance();
+        calendar.add(Calendar.DAY_OF_YEAR, -daysCount);
+        Date cutoffDate = calendar.getTime();
+
+        try (Session session = KruizeHibernateUtil.getSessionFactory().openSession()) {
+            List<String> tablenames = session.createNativeQuery(SELECT_ALL_KRUIZE_TABLES).getResultList();
+            for (String tableName : tablenames) {
+                String datePart = null;
+                if (tableName.startsWith("kruize_results_")) {
+                    datePart = tableName.substring("kruize_results_".length());
+                } else if (tableName.startsWith("kruize_recommendations_")) {
+                    datePart = tableName.substring("kruize_recommendations_".length());
+                }
+                if (null != datePart) {
+                    Date tableDate = new SimpleDateFormat("yyyyMMdd").parse(datePart);
+                    // Compare the date part with the cutoff date
+                    if (tableDate.after(cutoffDate)) {
+                        LOGGER.info("Table not eligible for deletion: " + tableName);
+                    } else {
+                        LOGGER.info("Table found for deletion: " + tableName);
+                        Transaction tx = session.beginTransaction();
+                        session.createNativeQuery("DROP TABLE " + tableName).executeUpdate();
+                        tx.commit();
+                    }
+                }
+            }
+        } catch (Exception e) {
+            LOGGER.error("Exception occurred while deleting the partition: {}", e.getMessage());
```

Comment:
Please use constants for all of the strings here

Reply:
What is the meaning of `cutoffDate`, what does that represent. Can you please add some explanation into the code.

Reply:
I'm concerned that there is no validation for `tableName` happening here. What happens if that is `null`? Also lets add some validation

Reply:
done